libdokan/folderlist.go

```diff
@@ -86,7 +93,15 @@ func (fl *FolderList) open(ctx context.Context, oc *openContext, path []string)
 		}
 		fl.mu.Unlock()
 
+		h, er := libkbfs.ParseTlfHandle(
```

Comment:
Oh I see.  I think it might be confusing to not check an error code immediately after receiving it.  For example, you probably shouldn't be calling `addToFavorite` in the `ok` case if `er` is not `nil` and is not the "canonical name" error.  Hopefully you can just eliminate the first `addToFavorite` call and not need this anymore.  But if you still need it, I would call it something more descriptive (`parseHandleErr`) and check it for fatal error conditions immediately.